sdk/pubkey/src/lib.rs

```diff
@@ -1249,4 +1255,26 @@ mod tests {
         );
         assert!(pubkey_from_seed_by_marker(&PDA_MARKER[1..]).is_ok());
     }
+
+    #[test]
+    fn test_pubkey_error_from_primitive_exhaustive() {
+        for variant in PubkeyError::iter() {
+            let variant_i64 = variant as i64;
+            assert_eq!(
+                PubkeyError::from_repr(variant_i64 as usize),
+                PubkeyError::from_i64(variant_i64)
+            );
```

Comment:
Let's also have a check for the `From<u64>` implementation
```suggestion
            assert_eq!(
                PubkeyError::from_repr(variant_i64 as usize),
                PubkeyError::from_i64(variant_i64)
            );
            assert_eq!(
                PubkeyError::from(variant_i64 as u64),
                variant
            );
```